service/controller/app/v1/resource/chart/desired.go

```diff
@@ -106,6 +109,31 @@ func hasSecret(cr v1alpha1.App, appCatalog v1alpha1.AppCatalog) bool {
 	return false
 }
 
+func replacePrefix(from string) string {
```

Comment:
So what are we using this for? Why do we replace the prefix? Which problem does that solve? I saw the comment in `processAnnotations` but I still don't understand it.

Reply:
All the annotations values we had put in `app` CR has prefix as `app-operator.giantswarm.io/` and we would like to change the value for chart-operator as it should manage that in TC. 

But as you questioned what it solves, it just changed the name but nothing. I have a suggestion maybe we could get rid of prefix at all, just using `giantswarm.io/`. 
so, for example, cordon-until key would be `giantswarm.io/cordon-until`
WDYT? @rossf7

Reply:
I think the approach is fine here. For context, we want to apply the annotations to the app CR not the chart CR. As the app CR is the public interface we expose both internally to release-operator and to customers.

So we want to translate the app-operator.* annotations to chart-operator.* and I think this is a good way to achieve that. We have the same use case for the version label.

Reply:
I like what Ji suggested towards the label naming, but you can decide that within your team.

Reply:
My 2c is when the annotation is supposed to be set by the user it probably makes sense to skip `*-operator` prefix.

Reply:
@kopiczko Late to the party but this isn't customer facing.

The cordon will be applied by our users via opsctl and programmatically by cluster-operator for the migration from chartconfig to app CRs. So I think using the usual prefix makes sense.